src/machine.tex

```diff
@@ -627,11 +627,13 @@ \subsection{Memory Privilege in {\tt mstatus} Register}
 
 The MPRV (Modify PRiVilege) bit modifies the privilege level at which
 loads and stores execute in all privilege modes.  When MPRV=0,
-translation and protection behave as normal.  When MPRV=1, load and
-store memory addresses are translated and protected as though the
-current privilege mode were set to MPP.  Instruction
-address-translation and protection are unaffected.  MPRV is hardwired
-to 0 if U-mode is not supported.
+translation and protection behave as normal, that is, M-mode accesses only go
+through the PMP mechanism (Section~\ref{sec:pmp}), and S-mode and U-mode
+accesses may also go through the address translation mechanism
+(Section~\ref{sv32algorithm}).  When MPRV=1, load and store memory addresses
```

Comment:
I don't think the MPRV description should explicitly mention any mechanisms like PMP or VM.  The main reason for this is composability: the MPRV text shouldn't depend on what protection/translation schemes exist.

Suppose someone adds a new custom extension for translation/protection.  MPRV should still work in the context of that mechanism, because all it does is change the effective privilege mode.

Reply:
I think I agree with you... As I was writing that, I was wondering about this.
However, I would like to have a way to easily know what "normal" find somewhere in the document an easy way to know what

> behave as normal

means. Doing this would break composability with newer mechanisms, but maybe giving an example would do the trick:
say a setup with MUS modes, no pmp, virtual memory and say that "behave as normal" in this case means translation happens for U / S and not M. This would clearly state that M-mode accesses don't go through mem translation... What do you think / where would this be appropriate in the document?

Reply:
Agreed...  "behaves as normal" is pretty vague, but was the best I could come up without abandoning generality.

Rather than clarifying this in the normative text, it might be best to explain the intended use of MPRV in a commentary block, and therein provide an example, which makes its effect more clear.  You want to give that a stab, or should I?